### PR TITLE
Improve, cleanup API-docs

### PR DESCRIPTION
**Pull request by mdellweg**
[noissue]

### CODE REVIEW

File: docs/plugin-writer/concepts/subclassing/serializers.rst [hunk 1]
```diff
@@ -34,7 +34,7 @@ serializer needs to add that field as well.
           )
 
       class Meta:
-            fields = ContentSerializer.Meta.fields + ('relative_path',)
+            fields = SingleArtifactContentSerializer.Meta.fields + ('relative_path',)
```
Comment (asmacdo) [inline docs/plugin-writer/concepts/subclassing/serializers.rst]: nice catch

**Comment by mdellweg** [comment]
@asmaco I updated the PR to the latest additions to the stages API and removed the WIP status.

File: docs/plugin-writer/concepts/tasks/add-remove.rst [hunk 2]
```diff
@@ -33,6 +33,10 @@ working directory setup, and database cleanup after encountering failures.
 Synchronizing
 -------------
 
+.. tip::
+
+    Please consider using the high level :ref:`stages-docs` for actual implementations.
```
Comment (asmacdo) [inline docs/plugin-writer/concepts/tasks/add-remove.rst]: I like this-- could you remove a similar section above?
https://github.com/pulp/pulpcore-plugin/pull/51/files#diff-6419377e4fd81d7840183ac963bd1047L6

File: pulpcore/plugin/stages/api.py [hunk 3]
```diff
@@ -230,6 +230,11 @@ class EndStage(Stage):
     """
 
     async def __call__(self):
+        """
+        This method drains items from the last queue and drops them.
+
+        Especially it does not try to put items into the nonexistent next queue.
```
Comment (asmacdo) [inline pulpcore/plugin/stages/api.py]: Optional nitpick: s/Especially/Importantly/

File: pulpcore/plugin/stages/association_stages.py [hunk 4]
```diff
@@ -91,7 +91,8 @@ class RemoveDuplicates(Stage):
     """
     Stage allows plugins to remove content that would break repository uniqueness constraints.
 
-    This stage is expected to be added by the DeclarativeVersion. See that class for example usage.
+    This stage is expected to be added by the
+    :class:`~pulpcore.plugin.stages.DeclarativeVersion`. See that class for example usage.
```
Comment (asmacdo) [inline pulpcore/plugin/stages/association_stages.py]: :+1:

**Comment by asmacdo** [comment]
@jortel, we aren't removing the second RemoveFuture anymore, so merging now.

**Pull request merged**